From 162d9b638915ec337807ac9806b02b98dfc60324 Mon Sep 17 00:00:00 2001
From: James Grant <james.grant@intel.com>
Date: Wed, 30 Aug 2023 14:23:12 +0100
Subject: [PATCH 1/4] Add missing soname to library, make symlink relative

---
 lib/makefile | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/lib/makefile b/lib/makefile
index 5d8c8bc..650baa9 100644
--- a/lib/makefile
+++ b/lib/makefile
@@ -12,7 +12,7 @@ CFLAGS += -fstack-protector-strong
 CFLAGS += -O2 -D_FORTIFY_SOURCE=2
 CFLAGS += -Wformat -Wformat-security
 
-LDFLAGS := -shared
+LDFLAGS := -shared -Wl,-soname,librsu.so.$(LIBRSU_VER)
 LDFLAGS += -z noexecstack
 LDFLAGS += -z relro -z now
 
@@ -20,7 +20,7 @@ all: librsu.so
 
 install: librsu.so
 	cp librsu.so $(INSTALL_PATH)/librsu.so.$(LIBRSU_VER)
-	ln -s $(INSTALL_PATH)/librsu.so.$(LIBRSU_VER) $(INSTALL_PATH)/librsu.so
+	ln -s librsu.so.$(LIBRSU_VER) $(INSTALL_PATH)/librsu.so
 
 librsu.so: $(SRC:.c=.o)
 	$(CROSS_COMPILE)gcc $(LDFLAGS) -o $@ $(SRC:.c=.o)
-- 
2.38.1

