From 43a021d73aa637cabdec699c1fcdd4ab0c2a3d97 Mon Sep 17 00:00:00 2001
From: Chang Rebecca Swee Fun <rebecca.swee.fun.chang@intel.com>
Date: Thu, 29 Oct 2020 21:20:10 +0800
Subject: [PATCH 4/4] dtbt: parse the file according to the text encoding

Signed-off-by: Chang Rebecca Swee Fun <rebecca.swee.fun.chang@intel.com>
---
 dtbt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/dtbt b/dtbt
index ff576fb..63d01b3 100755
--- a/dtbt
+++ b/dtbt
@@ -136,7 +136,7 @@ def configfs_status(ovl_dir):
         exit(1)
 
     # Read status file, removing trailing newlines, etc
-    f_status = open(cfs_full_path, "rb")
+    f_status = open(cfs_full_path, "r")
     status = f_status.read().rstrip()
     f_status.close()
 
-- 
2.17.1

